src/Features/DynamicCubemaps.cpp

```diff
@@ -91,11 +91,28 @@ void DynamicCubemaps::DrawSettings()
 					DirectX::ScratchImage image;
 					DX::ThrowIfFailed(CaptureTexture(device, context, tempTexture, image));
 
-					std::string filename = std::format("Data\\Textures\\DynamicCubemaps\\{:.2f}{:.2f}{:.2f}R{:.2f}.dds", settings.CubemapColor.x, settings.CubemapColor.y, settings.CubemapColor.z, settings.CubemapColor.w);
+					bool defaultDynamicCubeMapSavePathSafe = true;
+					try {
+						if (std::filesystem::create_directories(defaultDynamicCubeMapSavePath)) {
+							logger::info("Missing DynamicCubeMap Creator directory created: {}", defaultDynamicCubeMapSavePath);
+						}
+					} catch (const std::exception) {
+						logger::error("Failed to create missing DynamicCubeMap Creator directory: {}", defaultDynamicCubeMapSavePath);
+						defaultDynamicCubeMapSavePathSafe = false;
+					}
 
-					std::wstring wfilename = std::wstring(filename.begin(), filename.end());
-					DX::ThrowIfFailed(SaveToDDSFile(image.GetImages(), image.GetImageCount(), image.GetMetadata(), DirectX::DDS_FLAGS::DDS_FLAGS_NONE, wfilename.c_str()));
+					if (defaultDynamicCubeMapSavePathSafe) {
+						std::filesystem::path DynamicCubeMapSavePath = defaultDynamicCubeMapSavePath;
+						std::filesystem::path filename(std::format("R{:03d}G{:03d}B{:03d}A{:03d}.dds", colorPixel.r, colorPixel.g, colorPixel.b, colorPixel.a));
+						DynamicCubeMapSavePath /= filename;
 
+						if (std::filesystem::exists(DynamicCubeMapSavePath)) {
+							logger::info("DynamicCubeMap Creator file for {} already exists, skipping.", filename.string());
+						} else {
+							DX::ThrowIfFailed(SaveToDDSFile(image.GetImages(), image.GetImageCount(), image.GetMetadata(), DirectX::DDS_FLAGS::DDS_FLAGS_NONE, DynamicCubeMapSavePath.c_str()));
+							logger::info("DynamicCubeMap Creator file for {} written", filename.string());
```

Comment:
Do we need a try for saving errors?

Reply:
will try to force an error on that to see. 

I did not note DX::ThrowIfFailed in try blocks which I did a very quick look.

Reply:
I believe it is catchable, I'd try first with a filename that isn't writeable.

Reply:
related DX:ThrowIfFailed moved inside try/catch block